hpct-ldap-server/src/managers/openldap.py

```diff
@@ -0,0 +1,491 @@
+
+        Parameters
+        ----------
+        admin_passwd : str
+            LDAP password.
+        domain : str
+            Domain name.
+        org : str
+            Organization name.
+        """
+        if domain and org and admin_passwd:
+            dcs = self._split_domain(domain)
+
+            # Reconfigure slapd with password
+            self.slapd_config(admin_passwd, domain, org)
+            self._add_base(admin_passwd, dcs)
+        else:
+            raise Exception("Domain, Organization, or Password is missing cannot complete action.")
+
+    def disable(self) -> None:
+        """Disable services."""
+        for name in self.systemd_services:
+            systemd.service_pause(name)
+
+    def enable(self) -> None:
+        """Enable services."""
+        for name in self.systemd_services:
+            systemd.service_resume(name)
+
+    def install(self) -> None:
+        """Install packages."""
+        if self.packages:
+            try:
+                apt.update()
+                for name in self.packages:
+                    apt.add_package(name)
+            except:
+                raise Exception(f"failed to install package ({name})")
+
+    def is_enabled(self) -> bool:
+        """Check enabled status of services."""
+        if self.systemd_services:
+            for name in self.systemd_services:
+                if not systemd._systemctl("is-enabled", name, quiet=True):
+                    return False
+
+        return True
+
+    def is_installed(self) -> bool:
+        """Check packages are installed."""
+        if self.packages:
+            for name in self.packages:
+                if not apt.DebianPackage.from_installed_package(name).present:
+                    return False
+
+        return True
+
+    def is_running(self) -> bool:
+        """Check running/active status of services."""
+        if self.systemd_services:
+            for name in self.systemd_services:
+                if not systemd.service_running(name):
+                    return False
+
+        return True
+
+    def restart(self) -> None:
+        """Restart servers/services."""
+        self.stop()
+        self.start()
+
+    def slapd_config(self, admin_passwd, domain, org) -> None:
+        """Configuration of slapd noninteractive.
+
+        Parameters
+        ----------
+        admin_passwd : str
+            LDAP password.
+        domain : str
+            Domain name.
+        org : str
+            Organization name.
+        """
+        args = [
+            "slapd slapd/no_configuration boolean false",
+            f"slapd slapd/domain string {domain}",
+            f"slapd shared/organization string {org}",
+            f"slapd slapd/password1 password {admin_passwd}",
+            f"slapd slapd/password2 password {admin_passwd}",
+            "slapd slapd/purge_database boolean true",
+            "slapd slapd/move_old_database boolean true",
+        ]
+        args = "\n".join(args)
```

Comment:
With the "(" and ")" instead of "[" and "]", this would not require a `"\n".join()`.

Reply:
Updated as suggested